pytest-embedded-serial-esp/pytest_embedded_serial_esp/serial.py

```diff
@@ -172,27 +143,21 @@ def use_esptool(hard_reset_after: bool = True, no_stub: bool = False):
             hard_reset_after: run hard reset after
             no_stub: disable launching the flasher stub
         """
+        warn(
+            "The 'no_stub' parameter is now read directly from `flasher_args.json` "
+            'and does not need to be explicitly set.',
+            DeprecationWarning,
+        )
 
         def decorator(func):
             @functools.wraps(func)
             def wrapper(self, *args, **kwargs):
                 with self.disable_redirect_thread():
                     with contextlib.redirect_stdout(self._q):
                         settings = self.proc.get_settings()
-                        try:
-                            self.esp = esptool.detect_chip(self.proc, self.baud)
-                            self.esp.connect('hard_reset')
-
-                            if not no_stub:
-                                self.stub = self.esp.run_stub()
-
-                            ret = func(self, *args, **kwargs)
-                        finally:
-                            if hard_reset_after:
-                                self.esp.hard_reset()
-
-                            self.proc.apply_settings(settings)
-
+                        self.esp.connect()
```

Comment:
do we need it?

Reply:
After a reset without a connection, we are not able to do anything. We need to connect first; otherwise, esptool will raise an error like 'Received incorrect bytes'.